hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/PrehandleHandlerContext.java

```diff
@@ -29,63 +30,141 @@
 import edu.umd.cs.findbugs.annotations.Nullable;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
 
 /**
- * Base abstract class for constructing {@link TransactionMetadata} by collecting information that
- * is needed when transactions are handled as part of "pre-handle" needed for signature
- * verification.
- *
- * <p>NOTE : This class is designed to be subclassed
+ * Interface for constructing {@link TransactionMetadata} by collecting information that is needed
+ * when transactions are handled as part of "pre-handle" needed for signature verification.
  */
-public abstract class TransactionMetadataBuilder<T extends TransactionMetadataBuilder<T>> {
-    protected final List<HederaKey> requiredNonPayerKeys = new ArrayList<>();
-    protected HederaKey payerKey;
-    protected ResponseCodeEnum status = OK;
-    protected TransactionBody txn;
-    protected final AccountKeyLookup keyLookup;
-    protected AccountID payer;
-
-    protected TransactionMetadataBuilder(@NonNull final AccountKeyLookup keyLookup) {
-        this.keyLookup = Objects.requireNonNull(keyLookup);
+public class PrehandleHandlerContext {
```

Comment:
There is one class named `PreHandleContext` and this class named `PreHandlerContext` . Should we merge both ?

Reply:
I did not create `PrehandleContext` and therefore I was not sure, what the intention was for that class. Maybe it has become obsolete with the new design and we can happily repurpose it.

Reply:
I think we need the account numbers and file numbers in some prehandle methods. So I created that class. I think we can merge both. Need not be done in this PR a separate PR

Reply:
Will do